src/operators.jl

```diff
@@ -571,7 +571,7 @@ end; end
 
 # The following are primarily there for internal use in the macro code for @constraint
 for op in [:(+), :(-)]; @eval begin
-    function $op(lhs::Array{Variable},rhs::Array{Variable})
+    function $op(lhs::AbstractArray{Variable},rhs::AbstractArray{Variable})
         (sz = size(lhs)) == size(rhs) || error("Incompatible sizes for $op: $sz $op $(size(rhs))")
         ret = Array{AffExpr}(sz)
         for I in eachindex(ret)
```

Comment:
I'm concerned about the validity of using ``eachindex`` from ``ret`` to index into ``lhs`` when these could be different types.